pgmpy/factors/discrete/CI.py

```diff
@@ -0,0 +1,244 @@
+from itertools import product
+
+import numpy as np
+
+from pgmpy.factors.discrete import TabularCPD
+
+
+class BinaryInfluenceModel(TabularCPD):
+    """
+    Canonical influence model for binary child variables
+    (Noisy-OR / Noisy-AND / Leaky variations).
+
+    This model is equivalent to the classic "Noisy-OR" or "Noisy-AND"
+    canonical activation vector (CAV) model.
+
+    Parameters
+    ----------
+    variable : str
+        The name of the influenced variable (child node).
+
+    evidence : list of str
+        The list of parent nodes (causes) influencing the variable.
+
+    activation_magnitude : list or array-like
+        The activation probabilities for each evidence variable, i.e.
+        P(X=1 | parent_i=1, others inactive).
+
+    mode : {'OR', 'AND'}, default='OR'
+        The combination scheme:
+            - 'OR'  → Noisy-OR (independent causes for activation)
+            - 'AND' → Noisy-AND (all causes needed for activation)
+
+    leak : float, optional
+        Probability that the variable is activated spontaneously (without any active parent).
+
+    isboolean_style : bool, default=False
+        Whether to interpret states as boolean (`True`/`False`) or numeric (`1`/`0`).
+
+    Examples
+    --------
+    >>> cav = BinaryInfluenceModel(
+    ...     variable="Disease",
+    ...     evidence=["Fever", "Cough", "Fatigue"],
+    ...     activation_magnitude=[0.6, 0.4, 0.2],
+    ...     leak=0.05,
+    ...     mode="OR",
+    ... )
+    >>> # The created object is a TabularCPD
+    """
+
+    def __init__(
+        self,
+        variable,
+        evidence,
+        activation_magnitude,
+        mode="OR",
+        leak=None,
+        isboolean_style=False,
```

Comment:
Is there a use case in your mind where using True/False as state names wouldn't be sufficient? If not, I think it would be better to simplify the implementation and remove `isboolean_style` argument with the default being True/False.